hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java

```diff
@@ -1048,6 +1052,30 @@ else if ( fkTarget instanceof EmbeddableValuedModelPart ) {
 		}
 	}
 
+	private static void setReferencedEmbeddableForeignKeyDescriptor(
```

Comment:
IMO this method should be removed i.e. revert to using `buildEmbeddableForeignKeyDescriptor`. Either this method does the same as `buildEmbeddableForeignKeyDescriptor`, except for calling `setForeignKeyDescriptor`, or `buildEmbeddableForeignKeyDescriptor` is using the wrong selection mapping for the target columns and it should be fixed. Or am I missing something?

Reply:
it means to make tobuildEmbeddableForeignKeyDescriptor more complex, decide about the target of referenced columns and selections.

Reply:
in this case we can just inline it